src/nfv_test_api/v2/services/gnodeb.py

```diff
@@ -236,15 +236,30 @@ def node_status(self, identifier: str) -> Dict[str, Any]:
         ]
 
         status["pid"] = self.process_handler.processes[identifier].pid
+
+        # Load the logs from the file
+        with get_file_path(identifier, FileType.LOG).open(mode="r") as out:
+            status["logs"] = [line.rstrip("\n") for line in out]
+
+        return_code = self.process_handler.processes[identifier].poll()
+        if return_code is not None:
+            # If the process is still in self.process_handler but has terminated then it is a zombie process
+            self.process_handler.kill(identifier)  # handle zombie process
+            status["logs"].extend(
+                [
+                    f"The gnodeB process failed with return code {return_code}.",
+                    "Killing zombie process, next call to status will raise Not Found.",
```

Comment:
It will not raise NotFound, because the service is still in the processes dict.

Reply:
The kill command removes the process from the dict so it will raise not found at the next call.

Reply:
Oh, okay, sorry.  I thought the kill method was on the process object.  Should we kill it now? Why not just log the error and kill it when calling stop?

Reply:
I have no strong opinion about that . I was under the assumption that I should clean this zombie process as early as possible and not wait that the user calls stop to kill it.

Reply:
Done